src/components/overlay_mask/overlay_mask.test.tsx

```diff
@@ -83,11 +84,16 @@ describe('EuiOverlayMask', () => {
     });
   });
 
+  // React 18 is really unhappy when RTL wants to unmount
+  // it while the component is still updating.
```

Comment:
Let's link the issue we opened for this in a code comment
```suggestion
  // React 18 is really unhappy when RTL wants to unmount
  // it while the component is still updating.
  // TODO: https://github.com/elastic/eui/issues/6998
```

Reply:
Addressed in https://github.com/elastic/eui/pull/7012/commits/2fa936eeb565a1ef72ea85018fcd833a56f217f4